File: Tff.FSharp.CodeCamp2013/Tff.Ransom.VB/DollarProvider.vb
Public Class DollarProvider

    Public Function GetDollars(ByVal numberOfDollars As Int32) As List(Of Dollar)
        Dim dollars As New List(Of Dollar)()
        Dim currentDollar As Dollar = Nothing
        Dim random As New Random()
        For dollarIndex As Integer = 0 To numberOfDollars - 1
            currentDollar = New Dollar()
            currentDollar.FederalReserveDistrictNumber = random.Next(1, 13)
            currentDollar.Id = dollarIndex

            Dim serialNumer As String = String.Empty
            For serialNumberIndex As Integer = 0 To 9
                serialNumer += random.Next(0, 9).ToString()
            Next

            currentDollar.SerialNumber = serialNumer
            currentDollar.SeriesDate = 2000 + random.Next(0, 10)
            dollars.Add(currentDollar)
        Next
        Return dollars
    End Function

End Class

File: Tff.FSharp.CodeCamp2013/Tff.Ransom.CS/DollarProvider.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace Tff.Ransom.CS
{
    public class DollarProvider
    {
        public List<Dollar> GetDollars(Int32 numberOfDollars)
        {
            List<Dollar> dollars = new List<Dollar>();
            Dollar currentDollar = null;
            Random random = new Random();
            for (int i = 0; i < numberOfDollars; i++)
            {
                currentDollar = new Dollar();
                currentDollar.FederalReserveDistrict = random.Next(1, 13);
                currentDollar.Id = i;

                String serialNumber = String.Empty;
                for (int j = 0; j < 9; j++)
                {
                    serialNumber += random.Next(0, 9).ToString();
                }
                currentDollar.SerialNumber = serialNumber;
                currentDollar.SeriesDate = 2000 + random.Next(0, 10);
                dollars.Add(currentDollar);
            }
            return dollars;
        }
    }
}
